From 675b35b7ed416c837267e493b157167319e8f5fa Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Thu, 1 Sep 2022 11:01:53 -0700
Subject: [PATCH] configure: Check for readline() instead of main() in
 libreadline

while checking for presense of libreadline, poke for a function which is
provided by libreadline, main is not provided by it, so modern compiler
toolchains may complain about it.

Upstream-Status: Submitted [https://github.com/Quagga/quagga/pull/9]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index ed279f48..d444ab4a 100755
--- a/configure.ac
+++ b/configure.ac
@@ -754,7 +754,7 @@ dnl	 [TODO] on Linux, and in [TODO] on Solaris.
 	      )]
 	    )]
 	  )
-         AC_CHECK_LIB(readline, main, LIBREADLINE="-lreadline $LIBREADLINE",,
+         AC_CHECK_LIB(readline, readline, LIBREADLINE="-lreadline $LIBREADLINE",,
                       "$LIBREADLINE")
          if test $ac_cv_lib_readline_main = no; then
            AC_MSG_ERROR([vtysh needs libreadline but was not found and usable on your system.])
-- 
2.37.3

